lib/node_modules/@stdlib/array/empty-like/benchmark/benchmark.js

```diff
@@ -0,0 +1,317 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2023 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var isTypedArrayLike = require( '@stdlib/assert/is-typed-array-like' );
+var isArray = require( '@stdlib/assert/is-array' );
+var zeros = require( '@stdlib/array/zeros' );
+var pkg = require( './../package.json' ).name;
+var emptyLike = require( './../lib' );
+
+
+// MAIN //
+
+bench( pkg, function benchmark( b ) {
+	var arr;
+	var x;
+	var i;
+
+	x = zeros( 0, 'float64' );
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		arr = emptyLike( x );
+		if ( arr.length !== 0 ) {
+			b.fail( 'should have length 0' );
+		}
+	}
+	b.toc();
+	if ( !isTypedArrayLike( arr ) ) {
+		b.fail( 'should return a typed array' );
+	}
+	b.pass( 'benchmark finished' );
+	b.end();
+});
+
+bench( pkg+':dtype=float64', function benchmark( b ) {
+	var arr;
+	var x;
+	var i;
+
+	x = zeros( 0, 'float64' );
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		arr = emptyLike( x, 'float64' );
```

Comment:
```suggestion
		arr = emptyLike( x );
```

The intent of the benchmarks in this file is to measure dtype inference. By explicitly specifying the `dtype`, we, in theory, circumvent dtype inference which is not what we want.